src/python/pants/backend/core/tasks/prepare_resources.py

```diff
@@ -25,6 +27,12 @@ def register_options(cls, register):
     super(PrepareResources, cls).register_options(register)
     register('--confs', advanced=True, type=Options.list, default=['default'],
              help='Prepare resources for these Ivy confs.')
+    register('--resources-dir-patterns-for-jarring', advanced=True, type=Options.list, default=[],
+             help='Specify regex patterns of resources directories that should be zipped into '
+                  'jars.')
+    register('--short-resources-jar-path', advanced=True, action='store_true', default=True,
+             help='Replace the target id in a resource jar file path with a unique and randomized '
```

Comment:
It doesn't need to be randomized; just unique to the target. Making it stable would help with debugging... look around for `target.id hash` type stuff; pretty sure we're already doing that somewhere.